src/userbase-server/purge.js

```diff
@@ -0,0 +1,461 @@
+import uuidv4 from 'uuid/v4'
+import logger from './logger'
+import setup from './setup'
+import connection from './connection'
+import connections from './ws'
+import userController from './user'
+import appController from './app'
+import adminController from './admin'
+import { getMsUntil1AmPst } from './utils'
+
+const MS_IN_AN_HOUR = 60 * 60 * 1000
+const MS_IN_A_DAY = 24 * MS_IN_AN_HOUR
+const TIME_TO_PURGE = 30 * MS_IN_A_DAY
+
+const ddbWhileLoop = async (params, ddbQuery, action) => {
+  let itemsResponse = await ddbQuery(params)
+  let items = itemsResponse.Items
+
+  await action(items)
+
+  // can be optimized with parallel scan
+  while (itemsResponse.LastEvaluatedKey) {
+    params.ExclusiveStartKey = itemsResponse.LastEvaluatedKey
+    itemsResponse = await ddbQuery(params)
+    items = itemsResponse.Items
+
+    await action(items)
+  }
+}
+
+const permanentDeleteDeletedItems = async (items, closeConnectedClients, permanentDelete) => {
+  const permanentDeletePromises = []
+
+  for (let i = 0; i < items.length; i++) {
+    const item = items[i]
+
+    if (item['deleted']) {
+      // closing connected clients tries to prevent lingering clients from inserting data while purge is under way
+      closeConnectedClients(item)
+
+      if (new Date() - new Date(item['deleted']) > TIME_TO_PURGE) {
+        permanentDeletePromises.push(permanentDelete(item))
+      }
+    }
+  }
+
+  await Promise.all(permanentDeletePromises)
+}
+
+const scanForDeleted = async (TableName, closeConnectedClients, permanentDelete) => {
```

Comment:
This function does the deletion too, right? If so, maybe it should be called something else because right now it looks like it's only scanning.